src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test.Common/LanguageServerTestBase.cs

```diff
@@ -44,6 +49,48 @@ public LanguageServerTestBase()
 
         protected ILoggerFactory LoggerFactory { get; }
 
+        protected static async Task<(CSharpTestLspServer Server, Uri DocumentUri)> CreateCSharpTestLspServerAsync(
+            SourceText csharpSourceText,
+            ServerCapabilities serverCapabilities)
+        {
+            var clientCapabilities = new ClientCapabilities();
+            var workspace = new AdhocWorkspace();
```

Comment:
`TestWorkspace.Create`, supposedly creating AdhocWorkspace's in parallel is flaky IIRC